### DISCUSSION TITLE
How can I add my own Handlebars Helpers to SEQ?

### DISCUSSION THREAD

**Discussion by JHorvath-MaxetaTech** [datalust/seq-tickets#1757] [Help]
I tried putting them in a <script> block in the HTML template, but they didn't seem to take?

I wanted to create:

```
  <script>
    Handlebars.registerHelper('ifeq', function (a, b, options) {
      if (a == b) { return options.fn(this); }
      return options.inverse(this);
    });
    Handlebars.registerHelper('ifnoteq', function (a, b, options) {
      if (a != b) { return options.fn(this); }
      return options.inverse(this);
    });
  </script>
```

to do some string equivalency checks against property names like this:

```
  {{#each $Properties}}
    {{#ifeq @key "Source"}} 
      {{#each this.ContributingEvents}}
        {{#ifnoteq this.[2] "message"}} 
          {{this.[2]}}
        {{/ifnoteq}}
      {{/each}}
    {{/ifeq}}
  {{/each}}
```

So that I only display the event messages associated with an triggered alert in my notification e-mail.

**Comment by JHorvath-MaxetaTech** [comment]
Turns out there is already an existing helper in the form of `#if_eq`, I was able to find this as part of the default HTML template, so I ended up using this:

```
  {{#each $Properties}}
    {{#if_eq @key "Source"}} 
      {{#each this.ContributingEvents}}
        {{#if @index}} 
          {{this.[2]}}
        {{/if}}
      {{/each}}
    {{/if_eq}}
  {{/each}}
```

Which did the trick, but I couldn't seem to find any documentation about the default set of helpers available in SEQ.

Anyone know where those are listed (is they even are)?

P.S. I'd still like to know if we can write our own custom helpers, if possible.

**Reply by nblumhardt** [reply]
Hi John; the list of helpers is at:

https://github.com/datalust/seq-app-htmlemail#built-in-helper-functions

It's not possible to add them without forking the app, but forking is actually pretty straightforward if it's worthwhile (code and build scripts are in the same linked repo).

Hope this helps,
Nick
